cpp/src/cylon/net/ops/gather.cpp

```diff
@@ -153,6 +154,67 @@ cylon::Status cylon::mpi::Gather(const std::shared_ptr<cylon::TableSerializer> &
   return cylon::Status::OK();
 }
 
+cylon::Status cylon::mpi::GatherArrowBuffer(const std::shared_ptr<arrow::Buffer> &buf,
+                                            int gather_root,
+                                            const std::shared_ptr<cylon::CylonContext> &ctx,
+                                            std::vector<std::shared_ptr<arrow::Buffer>> &buffers) {
+
+  std::vector<int32_t> all_buffer_sizes;
+  if (AmIRoot(gather_root, ctx)) {
+    all_buffer_sizes.resize(ctx->GetWorldSize(), 0);
+  }
+
+  int32_t size = static_cast<int32_t>(buf->size());
+  int status = MPI_Gather(&size,
+                          1,
+                          MPI_INT32_T,
+                          all_buffer_sizes.data(),
+                          1,
+                          MPI_INT32_T,
+                          gather_root,
+                          MPI_COMM_WORLD);
+  if (status != MPI_SUCCESS) {
+    return cylon::Status(cylon::Code::ExecutionError, "MPI_Gather failed when receiving buffer sizes!");
+  }
+
+  std::shared_ptr<arrow::Buffer> all_buf = std::move(arrow::AllocateBuffer(0).MoveValueUnsafe());
```

Comment:
```suggestion
  std::shared_ptr<arrow::Buffer> all_buf;
```
we can leave the shared_ptr as it is (ie. `nullptr`)